kong/db/schema/init.lua

```diff
@@ -1085,7 +1085,9 @@ validate_fields = function(self, input)
       field, err = resolve_field(self, k, field, subschema)
       if field then
         _, errors[k] = self:validate_field(field, v)
-      else
+      elseif not (err == validation_errors.UNKNOWN and v == null) then
+        -- extra fields with value of null in the input config are ignored
+        -- otherwise record the error
```

Comment:
This should only happen when validating `declarative` configuration OR even stricter scope: when validating declarative config that control plane send to data plane over websocket.

Alternative way to implement this is to implement somekind of `remove_unknown_null_fields` BEFORE validating, then you don't need to change validation rules.